warp10/src/main/java/io/warp10/script/ext/http/HTTP.java

```diff
@@ -0,0 +1,533 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+
+package io.warp10.script.ext.http;
+
+import io.warp10.WarpConfig;
+import io.warp10.script.NamedWarpScriptFunction;
+import io.warp10.script.WarpScriptException;
+import io.warp10.script.WarpScriptStack;
+import io.warp10.script.WarpScriptStackFunction;
+import io.warp10.script.WebAccessController;
+import io.warp10.standalone.StandaloneWebCallService;
+import io.warp10.warp.sdk.Capabilities;
+
+import org.apache.commons.codec.binary.Base64;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * Send an HTTP request to a url
+ *
+ * To raise maximum number of calls and download size limit, use these capabilities:
+ * http.requests
+ * http.size
+ *
+ * Params:
+ * url The URL to send the request to. Must begin with http:// or https://
+ * method The optional http method. Default to GET
+ * headers An optional header map
+ * body An optional body. UTF-8 STRING or BYTES
+ * headers.macro A optional macro that expects this input parameters map on the stack, and push back the headers. Convenient for custom authorization schemes
+ * chunk.size Chunk size
+ * chunk.macro A macro that is executed whenever a chunk has been downloaded. It expects a MAP that contains chunk number (a LONG), status code (a LONG), status message (a STRING), headers (a MAP), and chunk content (a BYTES objects)
+ * username Optional field. If both username and password field are present and headers.macro is absent, basic authentication will be performed
+ * password Optional field. If both username and password field are present and headers.macro is absent, basic authentication will be performed
+ *
+ * Output:
+ * RESPONSE A map that contains status code (a LONG), status message (a STRING), headers (a MAP) and full content of the response (a BYTES objects). The content is empty if chunk option is used
+ *
+ */
+public class HTTP extends NamedWarpScriptFunction implements WarpScriptStackFunction {
+
+  //
+  // Arguments
+  //
+
+  public static final String METHOD = "method";
+  public static final String URL = "url";
+  public static final String HEADERS = "headers";
+  public static final String BODY = "body";
+  public static final String HEADERS_MACRO = "headers.macro";
```

Comment:
I'm, again, not convinced by the usefulness of this parameter. This does not offer more feature or readability than coding that in WarpScript:
```
{
  /* All your HTTP parameters */
}
@your_macro_updating_headers
HTTP
```